app/controllers/favorites_controller.rb

```diff
@@ -0,0 +1,32 @@
+class FavoritesController < ApplicationController
+  def index
+    @favorites = Favorite.includes(:post)
+  end
+
+  def create
+    @post = Post.find(params[:post_id])
+    @favorites = current_user.favorites.new(user_id: current_user.id, post_id: @post.id)
+
+    if @favorites.save
+      flash[:success] = 'Post favorited!'
+    else
+      flash[:error] = 'Post was not favorited!'
+    end
+
+    redirect_to favorites_path
+  end
+
+  def destroy
+    @favorite = Favorite.find(params[:id])
+
+    authorize(@favorite)
```

Comment:
Add simple Pundit to ensure that a user cannot unfavorite a post that was unfavorited by another user